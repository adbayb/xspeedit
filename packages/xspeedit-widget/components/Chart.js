import React, { PureComponent } from "react";
import PropTypes from "prop-types";

class Chart extends PureComponent {
	static propTypes = {
		title: PropTypes.string.isRequired,
		data: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.number))
	};

	colors = [
		"#FFF8E1",
		"#FFECB3",
		"#FFE082",
		"#FFD54F",
		"#FFCA28",
		"#FFC107",
		"#FFB300",
		"#FFA000",
		"#FF8F00",
		"#FF6F00"
	];

	renderObjects(objects) {
		return objects.map((object, index) => {
			return (
				object > 0 && (
					<div key={index} className="object">
						{object}
						<style jsx>{`
							.object {
								display: flex;
								flex: 1;
								color: black;
								font-size: 0.9rem;
								background-color: ${this.colors[object]};
								flex: none;
								height: ${object * 10}%;
								justify-content: center;
								align-items: center;
							}
						`}</style>
					</div>
				)
			);
		});
	}

	renderBoxes() {
		const { data } = this.props;

		return data.map((objects, index) => {
			return (
				<div key={index} className="box">
					{this.renderObjects(objects)}
					<style jsx>{`
						.box {
							display: flex;
							flex-direction: column;
							justify-content: flex-end;
							flex: none;
							width: 50px;
							margin-right: 10px;
							background-color: black;
							border: 3px solid black;
						}
					`}</style>
				</div>
			);
		});
	}

	render() {
		const { title } = this.props;

		return (
			<section>
				<h1>{title.toUpperCase()}</h1>
				<div>{this.renderBoxes()}</div>
				<style jsx>{`
					section {
						display: flex;
						flex-direction: column;
						flex: 1;
						padding: 20px 10px;
					}
					section:nth-child(2) {
						padding-bottom: 0;
					}
					h1 {
						margin: 0 0 20px 0;
						font-size: 12px;
						font-weight: bold;
					}
					div {
						display: flex;
						overflow-x: scroll;
						flex: 1;
					}
				`}</style>
			</section>
		);
	}
}

export default Chart;
