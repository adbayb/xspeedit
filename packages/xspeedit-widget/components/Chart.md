**Preview:**

```js
<Chart title="chart" data={[[1, 2], [3, 4]]} />;
```

**Example:**

```jsx static
import React from "react";

const Example = () => {
	return (
		<Chart title="chart" data={[[1, 2], [3, 4]]} />
	);
};
```
