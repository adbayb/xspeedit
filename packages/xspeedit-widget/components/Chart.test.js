import React from "react";
import Chart from "./Chart.js";

describe("Chart", () => {
	it("should render correctly", () => {
		// GIVEN: notre composant:
		const tree = shallow(<Chart title="Chart" data={[[1, 2], [3, 4]]} />);
		// THEN: il doit s'afficher correctement en matchant son snapshot:
		expect(tree).toMatchSnapshot();
	});
});
