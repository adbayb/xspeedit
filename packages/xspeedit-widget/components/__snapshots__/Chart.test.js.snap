// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart should render correctly 1`] = `
<section
  className="jsx-2374317178"
>
  <h1
    className="jsx-2374317178"
  >
    CHART
  </h1>
  <div
    className="jsx-2374317178"
  >
    <div
      className="jsx-1123166150 box"
      key="0"
    >
      <div
        className="jsx-2145737142 object"
        key="0"
      >
        1
        <JSXStyle
          css=".object.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:1;-ms-flex:1;flex:1;color:black;font-size:0.9rem;background-color:#FFECB3;-webkit-flex:none;-ms-flex:none;flex:none;height:10%;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}"
          dynamic={
            Array [
              "#FFECB3",
              10,
            ]
          }
          styleId="1993202757"
        />
      </div>
      <div
        className="jsx-1757275421 object"
        key="1"
      >
        2
        <JSXStyle
          css=".object.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:1;-ms-flex:1;flex:1;color:black;font-size:0.9rem;background-color:#FFE082;-webkit-flex:none;-ms-flex:none;flex:none;height:20%;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}"
          dynamic={
            Array [
              "#FFE082",
              20,
            ]
          }
          styleId="1993202757"
        />
      </div>
      <JSXStyle
        css=".box.jsx-1123166150{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:end;-webkit-justify-content:flex-end;-ms-flex-pack:end;justify-content:flex-end;-webkit-flex:none;-ms-flex:none;flex:none;width:50px;margin-right:10px;background-color:black;border:3px solid black;}"
        styleId="1123166150"
      />
    </div>
    <div
      className="jsx-1123166150 box"
      key="1"
    >
      <div
        className="jsx-981346144 object"
        key="0"
      >
        3
        <JSXStyle
          css=".object.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:1;-ms-flex:1;flex:1;color:black;font-size:0.9rem;background-color:#FFD54F;-webkit-flex:none;-ms-flex:none;flex:none;height:30%;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}"
          dynamic={
            Array [
              "#FFD54F",
              30,
            ]
          }
          styleId="1993202757"
        />
      </div>
      <div
        className="jsx-2081856332 object"
        key="1"
      >
        4
        <JSXStyle
          css=".object.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:1;-ms-flex:1;flex:1;color:black;font-size:0.9rem;background-color:#FFCA28;-webkit-flex:none;-ms-flex:none;flex:none;height:40%;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}"
          dynamic={
            Array [
              "#FFCA28",
              40,
            ]
          }
          styleId="1993202757"
        />
      </div>
      <JSXStyle
        css=".box.jsx-1123166150{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:end;-webkit-justify-content:flex-end;-ms-flex-pack:end;justify-content:flex-end;-webkit-flex:none;-ms-flex:none;flex:none;width:50px;margin-right:10px;background-color:black;border:3px solid black;}"
        styleId="1123166150"
      />
    </div>
  </div>
  <JSXStyle
    css="section.jsx-2374317178{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-flex:1;-ms-flex:1;flex:1;padding:20px 10px;}section.jsx-2374317178:nth-child(2){padding-bottom:0;}h1.jsx-2374317178{margin:0 0 20px 0;font-size:12px;font-weight:bold;word-break:break-word;}div.jsx-2374317178{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;overflow-x:scroll;-webkit-flex:1;-ms-flex:1;flex:1;}"
    styleId="2374317178"
  />
</section>
`;
